### DISCUSSION TITLE
Up and Down arrows are not the right way (osu!lazer mania mode)

### DISCUSSION THREAD

**Discussion by Zezinette** [ppy/osu#28373] [Q&A]
This is a copy and paste from a discussion I made in the official osu! forum. Thanks to "Blushing" for helping me resolve the problem by hinting me to post here.

Hi there! One of my friends showed me the new osu!lazer UI and I installed it. However, the mania skin I use in osu! (which had nothing wrong there) has a weird quirk where the up and down arrows are inverted.
From what I know it's the game not rotating the arrow notes, since the colors of the arrows are pink/green/pink/green, which is the case in both versions.
This bug didn't occur in normal osu! and from what I know hasn't been documented.
I tried unchecking the "upscroll" box but it ended up being even more glitched, as the problem was not resolved and the arrow notes were still upscrolling but the arrows themselves were at the bottom of the screen.
Any help would be greatly appreciated!

Screenshot here:
https://imgur.com/a/gIJsuJH

**Comment by bdach** [comment]
Please link the skin.

**Comment by Zezinette** [comment]
here is the skin :
https://osu.ppy.sh/community/forums/topics/512453?n=1
I'm using the Upscroll version

**Comment by bdach** [comment]
Oh dear god.

There's a whole bunch of stupid config nonsense in stable for up/downscroll.

Notably this is failing for the given skin because the skin version is 2.1, and for skins below 2.5 the sprites from the raw skin are never flipped even if upscroll is active:

```csharp
        internal bool GetNoteFlipVertical(int flipNumber, string flipPostfix = @"", bool defaultBool = true)
        {
            if (SkinManager.Current.Version < 2.5)
                return false;
            return GetFlipVertical(@"Note", flipNumber, flipPostfix, defaultBool);
        }
```

(https://github.com/peppy/osu-stable-reference/blob/0e425c0d525ef21353c8293c235cc0621d28338b/osu!/Graphics/Skinning/SkinMania.cs#L269-L274, also see https://osu.ppy.sh/wiki/en/Skinning/skin.ini#2.5)

so it's just completely broken because some of the sprites that stable wouldn't flip before 2.5 are flipped and things just break.

That aside, there's also a whole bunch of ini config keys (`UpsideDown`, `KeyFlipWhenUpsideDown`, `NoteFlipWhenUpsideDown` etc.) that control when the key/note images should be flipped in minute detail.

As far as @Zezinette is concerned - as far as I can tell you may as well use the downscroll variant of that specific skin and it'll be fine.

The open question to @ppy/team-client would be - do we bother to implement all of this crap? Do we even have a real sense of how many people out there in the wild are affected by this? I guess the fact that this never surfaced is proof enough that probably not much, but can we just not bother?

Is there someone else we can get some insight on this from? @rockroller01 perhaps?

**Reply by smoogipoo** [reply]
I think we do have to support this, if only to not break existing skins.

**Reply by bdach** [reply]
I'm very reluctant to because this is another multiple-tiered-lookup horrorshow with global stage skin settings, per-column-and-per-*element* skin settings, skin version all being involved in whatever process for determining whether a particular sprite should be flipped. Which is why I'm attempting to weasel out of having to support them if we absolutely don't have to (e.g. more than 5% skins out there in the wild use that stuff)

**Reply by Sikuromo** [reply]
I will say that, as a new user of lazer, this issue is also affecting me with an FNF skin. Are there any known fixes, or am I just going to have to pray that you'll fix it lol
